nodebuilder/tests/tastora/framework.go

```diff
@@ -0,0 +1,542 @@
+func configOverrides() toml.Toml {
+	overrides := make(toml.Toml)
+	txIndexConfig := make(toml.Toml)
+	txIndexConfig["indexer"] = "kv"
+	overrides["tx_index"] = txIndexConfig
+	return overrides
+}
+
+// getCelestiaTag returns the Celestia app image tag.
+func getCelestiaTag() string {
+	if tag := os.Getenv("CELESTIA_TAG"); tag != "" {
+		return tag
+	}
+	return defaultCelestiaTag
+}
+
+// getNodeTag returns the Celestia node image tag.
+func getNodeTag() string {
+	if tag := os.Getenv("CELESTIA_NODE_TAG"); tag != "" {
+		return tag
+	}
+	return defaultNodeTag
+}
+
+// getNodeImage returns the Celestia node image.
+func getNodeImage() string {
+	if tag := os.Getenv("CELESTIA_NODE_IMAGE"); tag != "" {
+		return tag
+	}
+	return nodeImage
+}
+
+// GenerateTestBlobs generates test blobs for blob operations
+func (f *Framework) GenerateTestBlobs(count int, dataSize int) ([]*blob.Blob, error) {
+	blobs := make([]*blob.Blob, count)
+
+	for i := 0; i < count; i++ {
+		// Create a test namespace (unique for each blob)
+		namespaceBytes := make([]byte, 10)
+		namespaceBytes[9] = byte(i + 1) // Make each namespace unique
+		namespace, err := libshare.NewV0Namespace(namespaceBytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create namespace: %w", err)
+		}
+
+		// Create test data
+		data := make([]byte, dataSize)
+		for j := range data {
+			data[j] = byte((i + j) % 256) // Fill with varying test data
+		}
+
+		// Create V0 blob (no signer required for test)
+		testBlob, err := blob.NewBlobV0(namespace, data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create test blob %d: %w", i, err)
+		}
+
+		blobs[i] = testBlob
+	}
+
+	return blobs, nil
+}
+
+// GenerateTestLibshareBlobs generates test libshare blobs for state API operations
+func (f *Framework) GenerateTestLibshareBlobs(count int, dataSize int) ([]*libshare.Blob, error) {
+	blobs := make([]*libshare.Blob, count)
+
+	for i := 0; i < count; i++ {
+		// Create a test namespace (unique for each blob)
+		namespaceBytes := make([]byte, 10)
+		namespaceBytes[9] = byte(i + 1) // Make each namespace unique
+		namespace, err := libshare.NewV0Namespace(namespaceBytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create namespace: %w", err)
+		}
+
+		// Create test data
+		data := make([]byte, dataSize)
+		for j := range data {
+			data[j] = byte((i + j) % 256) // Fill with varying test data
+		}
+
+		// Create V0 libshare blob (no signer required for test)
+		testBlob, err := libshare.NewV0Blob(namespace, data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create test libshare blob %d: %w", i, err)
+		}
+
+		blobs[i] = testBlob
+	}
+
+	return blobs, nil
+}
+
+// FillBlocks produces the given amount of contiguous blocks with customizable size.
+// This is needed for sync testing scenarios. Returns a channel that reports when done.
+func (f *Framework) FillBlocks(ctx context.Context, account string, blockSize, numBlocks int) (<-chan error, error) {
```

Comment:
This method seems completely useless. It only simulates block filling, but also does it quite inaccurately with a fixed 100ms duration per block. Instead, we have a handy WaitForHeight client method that renders this method useless.

cc @gupadhyaya to confirm if this is used in swamp ports to tastora.

Reply:
its used in the sync and das tests. let's revisit while review them. i prefer to keep it for now if ok?